change snackbar to a better maintained one

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: simple-zeeguu@0.1.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.x" from react-simple-snackbar@1.1.11
npm ERR! node_modules/react-simple-snackbar
npm ERR!   react-simple-snackbar@"^1.1.11" from the root project

--- a/package.json
+++ b/package.json
@@ -5,6 +5,7 @@
   "dependencies": {
     "@material-ui/core": "^4.11.4",
     "@material-ui/icons": "^4.11.2",
+    "@material-ui/lab": "^4.0.0-alpha.60",
     "@nivo/bar": "^0.72.0",
     "@nivo/core": "^0.72.0",
     "@nivo/line": "^0.72.0",
@@ -29,7 +30,6 @@
     "react-loader-spinner": "^4.0.0",
     "react-router-dom": "^5.2.0",
     "react-scripts": "^4.0.3",
-    "react-simple-snackbar": "^1.1.11",
     "remove-accents": "^0.4.2",
     "speak-tts": "^2.0.8",
     "styled-components": "^5.2.1",

--- a/src/exercises/Exercises.js
+++ b/src/exercises/Exercises.js
@@ -11,7 +11,6 @@ import LoadingAnimation from "../components/LoadingAnimation";
 import { setTitle } from "../assorted/setTitle";
 import Match from "./exerciseTypes/match/Match";
 import strings from "../i18n/definitions";
-import SnackbarProvider from "react-simple-snackbar";
 
 let BOOKMARKS_TO_PRACTICE = 10;
 
@@ -266,15 +265,13 @@ export default function Exercises({ api, articleID }) {
           toggleShow={toggleShow}
         />
       </s.ExForm>
-      <SnackbarProvider>
-        <FeedbackButtons
-          show={showFeedbackButtons}
-          setShow={setShowFeedbackButtons}
-          feedbackFunction={stopShowingThisFeedback}
-          currentExerciseType={currentExerciseType}
-          currentBookmarksToStudy={currentBookmarksToStudy}
-        />
-      </SnackbarProvider>
+      <FeedbackButtons
+        show={showFeedbackButtons}
+        setShow={setShowFeedbackButtons}
+        feedbackFunction={stopShowingThisFeedback}
+        currentExerciseType={currentExerciseType}
+        currentBookmarksToStudy={currentBookmarksToStudy}
+      />
     </s.ExercisesColumn>
   );
 }

--- a/src/exercises/FeedbackButtons.js
+++ b/src/exercises/FeedbackButtons.js
@@ -1,7 +1,12 @@
 import * as s from "./FeedbackButtons.sc.js";
 import { useState, useEffect } from "react";
 import strings from "../i18n/definitions";
-import { useSnackbar } from "react-simple-snackbar";
+import Snackbar from "@material-ui/core/Snackbar";
+import MuiAlert from "@material-ui/lab/Alert";
+
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
 
 export default function FeedbackButtons({
   show,
@@ -11,7 +16,6 @@ export default function FeedbackButtons({
   currentBookmarksToStudy,
 }) {
   const MATCH_EXERCISE_TYPE = "Match_three_L1W_to_three_L2W";
-  const SNACK_DURATION = 4500;
 
   const buttons = [
     { name: strings.bookmarkTooEasy, value: "too_easy" },
@@ -31,8 +35,8 @@ export default function FeedbackButtons({
   const [className, setClassName] = useState("");
   const [input, setInput] = useState("");
   const [selectedId, setSelectedId] = useState(null);
-
-  const [openSnackbar, closeSnackbar] = useSnackbar(s.SnackbarOptions);
+  const [feedback, setFeedback] = useState("");
+  const [openSnackbar, setOpenSnackbar] = useState(false);
 
   useEffect(() => {
     if (currentExerciseType !== MATCH_EXERCISE_TYPE) {
@@ -41,7 +45,7 @@ export default function FeedbackButtons({
       setSelectedId(null);
     }
     console.log(selectedId);
-    closeSnackbar();
+    setOpenSnackbar(false);
     setInput("");
     setShowInput(false);
     setClassName("");
@@ -54,14 +58,10 @@ export default function FeedbackButtons({
     } else {
       if (value !== "other") {
         feedbackFunction(value, selectedId);
-        let feedback = "";
         buttons.forEach((button) => {
-          if (button.value === value) feedback = button.name;
+          if (button.value === value) setFeedback(button.name);
         });
-        openSnackbar(
-          `${strings.sentFeedback1} "${feedback}" ${strings.sentFeedback2}`,
-          SNACK_DURATION
-        );
+        setOpenSnackbar(true);
         setShow(false);
         if (currentExerciseType === MATCH_EXERCISE_TYPE) {
           setSelectedId(null);
@@ -95,21 +95,26 @@ export default function FeedbackButtons({
         .replace(re2, "")
         .replaceAll(" ", "_");
       feedbackFunction(newFeedback, selectedId);
-      let feedback = input;
+      setFeedback(input);
       setInput("");
       setShowInput(false);
       setClassName("");
       if (currentExerciseType === MATCH_EXERCISE_TYPE) {
         setSelectedId(null);
       }
-      openSnackbar(
-        `${strings.sentFeedback1} "${feedback}" ${strings.sentFeedback2}`,
-        SNACK_DURATION
-      );
+      setOpenSnackbar(true);
       setShow(false);
       event.preventDefault();
     }
   }
+
+  const handleSnackbarClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setOpenSnackbar(false);
+  };
 
   return (
     <s.FeedbackHolder>
@@ -166,6 +171,19 @@ export default function FeedbackButtons({
           <s.FeedbackSubmit type="submit" value="Submit" />
         </s.FeedbackForm>
       )}
+      <Snackbar
+        anchorOrigin={{
+          vertical: "bottom",
+          horizontal: "right",
+        }}
+        open={openSnackbar}
+        autoHideDuration={4500}
+        onClose={handleSnackbarClose}
+      >
+        <Alert onClose={handleSnackbarClose} severity="success">
+          {`${strings.sentFeedback1} "${feedback}" ${strings.sentFeedback2}`}
+        </Alert>
+      </Snackbar>
     </s.FeedbackHolder>
   );
 }

--- a/src/exercises/FeedbackButtons.sc.js
+++ b/src/exercises/FeedbackButtons.sc.js
@@ -95,20 +95,6 @@ const FeedbackSubmit = styled.input`
   outline: none;
 `;
 
-const SnackbarOptions = {
-  position: "bottom-right",
-  style: {
-    backgroundColor: "#00665C",
-    color: "#ffffff",
-    textAlign: "center",
-    minWidth: "fit-content",
-    fontFamily: "",
-  },
-  closeStyle: {
-    marginLeft: "-1em",
-  },
-};
-
 export {
   FeedbackHolder,
   FeedbackButtonsHolder,
@@ -119,5 +105,4 @@ export {
   FeedbackForm,
   FeedbackInput,
   FeedbackSubmit,
-  SnackbarOptions,
 };

